### PR TITLE
Fix!: Use LONGTEXT data type when storing large blobs of text in MySQL

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

File: sqlmesh/core/state_sync/engine_adapter.py [hunk 1]
```diff
@@ -103,17 +104,20 @@ def __init__(
...
+        index_type = index_text_type(self.engine_adapter.dialect)
+        blob_type = blob_text_type(self.engine_adapter.dialect)
+
         self._snapshot_columns_to_types = {
-            "name": exp.DataType.build("text"),
-            "identifier": exp.DataType.build("text"),
-            "version": exp.DataType.build("text"),
-            "snapshot": exp.DataType.build("text"),
-            "kind_name": exp.DataType.build("text"),
+            "name": exp.DataType.build(index_type),
```
Comment (izeigerman) [inline sqlmesh/core/state_sync/engine_adapter.py:111]: @GeorgeSittas any reason we didn't update this before?

Comment (tobymao) [reply sqlmesh/core/state_sync/engine_adapter.py:111]: update what?

**Comment by tobymao** [comment]
i'm fine with what you've done here, but you do understand that this isn't a proper migration because old systems won't change right. the proper way would have been to make a new migration to change the types

File: sqlmesh/core/state_sync/engine_adapter.py (hunk 1 above)
Comment (GeorgeSittas) [reply sqlmesh/core/state_sync/engine_adapter.py:111]: Hmm I'm not fully sure - maybe an oversight?

**Comment by izeigerman** [comment]
@tobymao yes, you're right. Going to add a migration script for that

**Comment by tobymao** [comment]
then you should revert all the changes to the previous migrations

**Review by georgesittas — COMMENTED** [review]
The index type changes in the engine_adapter, plan modules look good, might have been an oversight on my end (at least for the engine adapter).

**Pull request merged**